### PR TITLE
Add make e2e test to dpu operator

### PR DESCRIPTION
**Pull request by vrindle**
Do not merge yet as this PR has this PR does rely on changes to [CDA and the spreadsheet](https://github.com/bn222/cluster-deployment-automation/pull/227) (those changes are up but not merged yet).

### CODE REVIEW

File: hack/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+CDA_REPO=${CDA_REPO:-https://github.com/bn222/cluster-deployment-automation.git}
+CDA_BRANCH=${CDA_BRANCH:-main}
+
+git submodule init
+git submodule update
+
+cd cluster-deployment-automation
+git checkout "$CDA_BRANCH"
+
+sed -e 's/#\(user\|group\) = ".*"$/\1 = "root"/' -i /etc/libvirt/qemu.conf
```
Comment (bn222) [inline hack/prepare.sh:12]: This has been moved into cda. Can you also update the README.md in cda ?

Comment (vrindle) [reply hack/prepare.sh]: I removed it... now we can merge and we can open a pr to update readme in cda.

File: hack/ipu_host_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+cd cluster-deployment-automation
+source ocp-venv/bin/activate
+
+export PATH=$PATH:"/usr/local/go/bin"
+
+# We need to overwrite the buildID to ensure that Jenkin's ProcessTreeKiller does not clean up the containers spawned by CDA on job completion. https://wiki.jenkins.io/display/JENKINS/ProcessTreeKiller
+# Specifically, we want the local container registry to persist so we do not need to rebuild images each time we run a new job.
+export BUILD_ID=dontKillMe
```
Comment (SalDaniele) [inline hack/ipu_host_deploy.sh:8]: I don't think we need this now...this is a Jenkins specific dial to manage how Jenkins handles process started by jobs...if we are decoupled from Jenkins with this change, this is not required.

**Comment by vrindle** [comment]
@SalDaniele this has been removed

**Pull request merged**